processing/src/main/java/org/apache/druid/segment/join/HashJoinSegment.java

```diff
@@ -50,15 +49,9 @@ public HashJoinSegment(
     this.baseSegment = baseSegment;
     this.clauses = clauses;
 
-    // Verify no clauses would shadow the special __time field.
-    for (JoinableClause clause : clauses) {
-      if (clause.includesColumn(ColumnHolder.TIME_COLUMN_NAME)) {
```

Comment:
Does this mean we now allow joins on the time column?

Reply:
Yes, we do, although that's not what this means. This same check is still being applied, but in the constructor of JoinableClause (it calls `Joinables.validatePrefix`).